engine/src/main/java/com/agiletec/aps/system/services/category/CategoryManager.java

```diff
@@ -126,16 +84,17 @@ public void addCategory(Category category) throws ApsSystemException {
 	 */
 	@Override
 	public void deleteCategory(String code) throws ApsSystemException {
-		Category cat = (Category) this.getCategories().get(code);
+		Category cat = this.getCategory(code);
 		if (cat != null && cat.getChildrenCodes().length <= 0) {
 			try {
 				this.getCategoryDAO().deleteCategory(code);
+				this.getCacheWrapper().deleteCategory(code);
 			} catch (Throwable t) {
 				_logger.error("Error detected while removing the category {}", code, t);
 				throw new ApsSystemException("Error detected while removing a category", t);
 			}
 		}
-		this.loadCategories();
+		this.initCache();

```

Comment:
what does the initCache() call do after a delete? Looks like the cache us updated in the try{} block above.

Reply:
The initCache() rebuilds all the categories because some categories may still have references with the deleted category (in the property "childrenCodes" or in the "parent").

Reply:
Ok makes sense. In the future (not this PR) do you think we will need to make those references concrete to the instance so they can be updated directly without a full cache rebuild?